src/components/TradeWidget/index.tsx

```diff
@@ -425,6 +425,28 @@ const preprocessTokenAddressesToAdd = (addresses: (string | undefined)[], networ
   return tokenAddresses
 }
 
+interface ChooseTokenInput {
+  tokens: TokenDetails[]
+  token: TokenDetails | null
+  tokenSymbol?: string
+  defaultTokenSymbol: 'DAI' | 'USDC'
+}
+
+const chooseTokenWithFallback = ({
+  tokens,
+  token,
+  tokenSymbol,
+  defaultTokenSymbol,
+}: ChooseTokenInput): TokenDetails => {
+  return (
+    token ||
+    (tokenSymbol && isAddress(tokenSymbol?.toLowerCase())
```

Comment:
~Why check if `tokenSymbol` is an address? Typo?~

Ohhh, because that's what's coming from the URL.

Can you maybe rename this var to something less confusing? Like `tokenIdentifierFromUrl` (too long) or `identifier` or `tokenSymbolFromUrl`. Hard to find a good name...